src/Engine/System/RenderSystem.cpp

```diff
@@ -111,4 +119,50 @@ void RenderSystem::update(float dt) {
 
 void RenderSystem::setCamera(const CameraComponent * camera) {
     s_camera = camera;
+}
+
+void RenderSystem::initFBO() {
+    // Initialize framebuffer to draw into
+    glGenFramebuffers(1, &s_fbo);
+    glBindFramebuffer(GL_FRAMEBUFFER, s_fbo);
+
+    glm::ivec2 size = Window::getFrameSize();
+
+    // Set the first texture unit as active
+    glActiveTexture(GL_TEXTURE0);
+
+    // Attach color to the framebuffer
+    glGenTextures(1, &s_fboColorTex);
+    glBindTexture(GL_TEXTURE_2D, s_fboColorTex);
+
+    glTexImage2D(
+        GL_TEXTURE_2D, 0, GL_RGB, size.x, size.y, 0, GL_RGB, GL_UNSIGNED_BYTE, NULL
+    );
+
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+
+    glFramebufferTexture2D(
+        GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D, s_fboColorTex, 0
+    );
+
+    // Create Renderbuffer Object to hold depth and stencil buffers
+    GLuint fboDepthRB(0);
+    glGenRenderbuffers(1, &fboDepthRB);
+    glBindRenderbuffer(GL_RENDERBUFFER, fboDepthRB);
+    glRenderbufferStorage(GL_RENDERBUFFER, GL_DEPTH24_STENCIL8, size.x, size.y);
+    glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_DEPTH_STENCIL_ATTACHMENT, GL_RENDERBUFFER, fboDepthRB);
+    // Using a texture instead
+    //glGenTextures(1, &depthTexture);
+    //glBindTexture(GL_TEXTURE_2D, depthTexture);
+    //glTexImage2D(GL_TEXTURE_2D, 0, GL_DEPTH_COMPONENT, size.x, size.y, 0, GL_DEPTH_COMPONENT, GL_FLOAT, 0);
+    //glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D, depthTexture, 0);
+
+    glBindFramebuffer(GL_FRAMEBUFFER, 0);
+    glBindTexture(GL_TEXTURE_2D, 0);
+}
+
+void RenderSystem::doResize() {
+    glViewport(0, 0, Window::getFrameSize().x, Window::getFrameSize().y);
+    initFBO();
```

Comment:
We don't need to reinitialize the entire FBO here. We just need to resize the texture it is bound to.

Reply:
You'd also need to resize the depth texture/renderbuffer. Apparently it's faster to just create a new FBO than resize all its attachments. I guess the standard best practice is actually to have a dummy FBO and swap between them, but since resizing is a very rare occurrence, I think this is fine

Reply:
Whatever you say boss man